compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ProvidesMethodFactoryGenerator.kt

```diff
@@ -110,6 +112,9 @@ internal class ProvidesMethodFactoryGenerator : PrivateCodeGenerator() {
 
     val isProperty = declaration is KtProperty
 
+    val isMangled = !isProperty &&
+      declaration.visibilityModifierTypeOrDefault() == INTERNAL_KEYWORD
```

Comment:
If the declaration is also marked as `@PublishedApi` then I _believe_ the name isn't mangled. May want to add a test for that just to make sure.

Reply:
Nice, pretty sure that the test case for that will fail at the moment. I couldn't find an API in the compiler to get the mangled function name, only for inline classes. This solution here is so fragile.